### PR TITLE
Implement new API endpoint /v1/users/me

### LINKED ISSUE #1596: an endpoint that gets current user's profile
Add an HTTP endpoint that displays detailed profile information about the current user. 
The profile should contain: 
 * id 
 * name
 * email
 * creationTime 
 * the list of project the user belongs to along with the group names.

Potential path could be of the form:

* /api/v1/me
* /api/v1/users

### To Do
- [x] implement an endpint
- [x] add unit test

child of #966

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:

This PR provides a new HTTP API endpoint, `/api/v1/users/me`, which returns user/project information about the current user.

**Which issue(s) this PR fixes**:

Fixes #1596

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/routes_v1.go [hunk 1]
```diff
@@ -1089,3 +1095,25 @@ func (r Routing) addUserToProject() http.Handler {
...
+// swagger:route GET /api/v1/users/me users getCurrentUser
+//
+// Returns information about the current user.
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       200: User
```
Comment (p0lyn0mial) [inline api/pkg/handler/routes_v1.go:1108]: We usually also add `401` - unauthorised.

File: api/pkg/handler/user.go [hunk 2]
```diff
@@ -93,6 +93,14 @@ func addUserToProject(projectProvider provider.ProjectProvider, userProvider pro
 	}
 }
 
+func getCurrentUserEndpoint(users provider.UserProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		authenticatedUser := ctx.Value(userCRContextKey).(*kubermaticapiv1.User)
+
+		return convertInternalUserToExternal(authenticatedUser), nil
```
Comment (p0lyn0mial) [inline api/pkg/handler/user.go:100]: wow, only two lines :)

File: api/pkg/handler/user_test.go [hunk 3]
```diff
@@ -579,3 +579,121 @@ func TestAddUserToProject(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCurrentUser(t *testing.T) {
```
Comment (p0lyn0mial) [inline api/pkg/handler/user_test.go:583]: thanks for the test, look really good.

File: api/pkg/handler/routes_v1.go [hunk 4]
```diff
@@ -181,6 +181,12 @@ func (r Routing) RegisterV1(mux *mux.Router) {
 	mux.Methods(http.MethodPost).
 		Path("/projects/{project_id}/users").
 		Handler(r.addUserToProject())
+
+	//
+	// Defines an endpoint to retrieve information about the current token owner
+	mux.Methods(http.MethodGet).
+		Path("/users/me").
```
Comment (p0lyn0mial) [inline api/pkg/handler/routes_v1.go]: I know that when we spoke the last time we agreed on the path. 

Now I'm starting to think that having `users` in the path implies that there might be an endpoint for retrieving users which to the best of my knowledge is not true. 

We will have endpoint that will list users of a project but then the path will be different.

What do you think ? Wouldn't be better to change the path to `/api/v1/me` ?

**Comment by p0lyn0mial** [comment]
@xrstf nice pull, only one minor comment.

File: api/pkg/handler/user.go (hunk 2 above)
Comment (xrstf) [reply api/pkg/handler/user.go:100]: All thanks to your work. :)

File: api/pkg/handler/routes_v1.go (hunk 4 above)
Comment (xrstf) [reply api/pkg/handler/routes_v1.go]: If there are no plans for a general list of users (`/api/v1/users`), then I'm fine with changing it to `/api/v1/me`.

**Review by p0lyn0mial — DISMISSED** [review]

**Comment by p0lyn0mial** [comment]
I think we also need to update the path in the tests
```
FAIL: TestGetCurrentUser (1.00s)
    --- FAIL: TestGetCurrentUser/scenario_1:_a_user_with_no_projects_yet_should_omit_the_`projects`_key_from_the_response (0.50s)
    	user_test.go:694: Expected HTTP status code 200, got 404: 404 page not found
    --- FAIL: TestGetCurrentUser/scenario_2:_a_user_who_is_assigned_to_projects (0.50s)
    	user_test.go:694: Expected HTTP status code 200, got 404: 404 page not found
```

**Review by p0lyn0mial — APPROVED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**